build.xml

```diff
@@ -59,31 +56,32 @@
 		<echo message="	manifest				 (Re)create the manifest file"/>
 		<echo message="	help						 Show this help"/>
 		<echo message="	jdoc						 Create JavaDOC"/>
-		<echo message="	release					Create release zip file"/>
+		<echo message="	release					Create release zip file based on a git tag"/>
 		<echo message="	zip							Create non-release zip file"/>
 		<echo message=""/>
 	</target>
 
-	<!-- Add build timestamp to version -->
-	<target name="-buildversion" unless="isRelease" >
+	<!-- Determine version -->
+	<target name="-version">
+		<exec executable="git" outputproperty="version.git">
+			<arg line="describe --tags --always --dirty"/>
+		</exec>
 		<tstamp>
-			<format property="version.build" pattern="yyyyMMddHHmmss" />
+			<format property="version.time" pattern="yyyyMMddHHmmss" />
 		</tstamp>
+		<hostinfo prefix="host"/>
+		<condition property="version" value="${version.git}-${version.time}" else="${version.git}">
+			<isfalse value="${isRelease}"/>
+		</condition>
 		<mkdir dir="${version.dest.dir}"/>
-		<propertyfile file="${version.dest.dir}/${version.build.file}">
-			<entry key="build" value="${version.build}"/>
+		<propertyfile file="${version.dest.dir}/${version.release.file}">
+			<entry key="release" value="${version}"/>
+			<entry key="release.user" value="${user.name}"/>
+			<entry key="release.time" value="${version.time}"/>
+			<entry key="release.host" value="${host.NAME}"/>
```

Comment:
On my machine this gives `localhost` despite a hostname being set.

Reply:
The logic is per https://ant.apache.org/manual/Tasks/hostinfo.html. It's something, the username will likely be more useful.